pkg/storegateway/indexheader/index/postings.go

```diff
@@ -546,7 +556,12 @@ func (t *PostingOffsetTableV2) LabelValuesOffsets(name, prefix string, filter fu
 		nextEntry pEntry
 	)
 
+	count := 1
 	for d.Err() == nil && !currEntry.isLast {
+		if count%checkContextEveryNIterations == 0 && ctx.Err() != nil {
+			return nil, ctx.Err()
```

Comment:
can we wrap the error so we have a bit more details? Also using `context.Cause` instead of `ctx.Err()` would give even more details